shell/platform/darwin/macos/framework/Source/FlutterPlatformNodeDelegateMac.mm

```diff
@@ -0,0 +1,105 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterPlatformNodeDelegateMac.h"
+
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterAppDelegate.h"
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterEngine_Internal.h"
+
+#include "flutter/shell/platform/common/accessibility_bridge.h"
+#include "flutter/third_party/accessibility/ax/ax_action_data.h"
+#include "flutter/third_party/accessibility/ax/ax_node_position.h"
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node.h"
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node_base.h"
+#include "flutter/third_party/accessibility/base/string_utils.h"
+#include "flutter/third_party/accessibility/gfx/geometry/rect_conversions.h"
+
+namespace flutter {  // namespace
+
+FlutterPlatformNodeDelegateMac::FlutterPlatformNodeDelegateMac(FlutterEngine* engine)
+    : engine_(engine) {
+  ax_platform_node_ = ui::AXPlatformNode::Create(this);
+  NSCAssert(ax_platform_node_, @"Failed to create platform node.");
+}
+
+FlutterPlatformNodeDelegateMac::~FlutterPlatformNodeDelegateMac() {
+  ax_platform_node_->Destroy();
+}
+
+gfx::NativeViewAccessible FlutterPlatformNodeDelegateMac::GetNativeViewAccessible() {
+  NSCAssert(ax_platform_node_, @"Platform node does not exist.");
+  return ax_platform_node_->GetNativeViewAccessible();
+}
+
+gfx::NativeViewAccessible FlutterPlatformNodeDelegateMac::GetParent() {
+  gfx::NativeViewAccessible parent = FlutterPlatformNodeDelegate::GetParent();
+  if (!parent) {
+    NSCAssert(engine_, @"Flutter engine should not be deallocated");
+    return engine_.viewController.view;
+  }
+  return parent;
+}
+
+gfx::Rect FlutterPlatformNodeDelegateMac::GetBoundsRect(
+    const ui::AXCoordinateSystem coordinate_system,
+    const ui::AXClippingBehavior clipping_behavior,
+    ui::AXOffscreenResult* offscreen_result) const {
```

Comment:
This method feels pretty busy. Logically it's:
1. Get flutter bounds rect.
2. Convert to macos bounds rect.
3. Map to view co-ordinates.
4. Map to screen co-ordinates.

Consider breaking this up a little bit for readability.